### ISSUE TITLE
The aspect ratio of Placeholder and ImageView is different, which leads to display BUG

### ISSUE THREAD

**Issue by Khaos116** [coil-kt/coil#607]
Use Placeholder 
![image](https://user-images.githubusercontent.com/11989916/101447489-eb723300-3957-11eb-9859-f58beea83bb3.png)
No Placeholder 
![image](https://user-images.githubusercontent.com/11989916/101447767-718e7980-3958-11eb-9670-a96cb5a4b2a7.png)

**Comment by caiyoufei** [comment]
Version is 1.1.0

**Comment by colinrtwhite** [comment]
@caiyoufei It's not clear what the bug and expected behaviour here is. Please include a sample app that reproduces the issue. I would also ensure your `ConstraintLayout` version is the latest as there have been a number of patch releases with sizing fixes.

**Comment by caiyoufei** [comment]
Strange, I can’t reproduce it today, I temporarily closed it
Thanks very much

**Comment by caiyoufei** [comment]
I found the cause of this problem, because layer-list  was used as a placeholder。Demo url is :
https://github.com/caiyoufei/CoilHolderBug

**Comment by caiyoufei** [comment]
![image](https://user-images.githubusercontent.com/11989916/101876115-7a808480-3bbe-11eb-90b0-9b6e7ec0c336.png)

**Comment by caiyoufei** [comment]
![image](https://user-images.githubusercontent.com/11989916/101876578-2cb84c00-3bbf-11eb-9a8f-c5c877cb5744.png)

**Comment by colinrtwhite** [comment]
Thanks for the sample! I'll take a look at it this week.

**Comment by destructo570** [comment]
I am facing this issue as well. After scrolling off the image from the screen and scrolling back to that image shows the correct aspect ratio. For now I have disable placeholder image. Is there any fix planned for this?

**Comment by colinrtwhite** [comment]
Still figuring out the best way to address this without unexpected behaviour changes for existing users. In the meantime to work around the issue set `precision(Precision.EXACT)` on either your request or your image loader.

**Comment by coconut85** [comment]
Facing the same issue as well. ImageView with `android:adjustViewBounds="true"` and `android:scaleType="fitXY"` in a RecyclerView with a placeholder. ImageView keeps the placeholder aspect ratio.

Tried to add `precision(Precision.EXACT)` and `precision(Precision.EXACT)` but didn't fix.
Works well without a placeholder. 

```
<ImageView
  android:layout_marginTop="10dp"
  android:adjustViewBounds="true"
  android:layout_gravity="center_horizontal"
  android:id="@+id/ivGif"
  android:scaleType="fitXY"
  android:layout_width="match_parent"
  android:layout_height="wrap_content"/>
```


```
view.ivGif.load(File(gifPath)) {
  crossfade(true)
  size(OriginalSize)
  precision(Precision.EXACT)
  placeholder(R.drawable.gif_loading_placeholder)
}
```

**Comment by yaseen2591** [comment]
I'm experiencing a same issue as well. sadly  `precision(Precision.EXACT)` didn't help me to fix the issue.

 ```
  <ImageView
                    android:layout_width="0dp"
                    android:layout_height="0dp"
                    android:scaleType="fitXY"
                    android:id="@+id/view_gif"
                    app:layout_constraintStart_toStartOf="parent"
                    app:layout_constraintTop_toTopOf="parent"
                    app:layout_constraintEnd_toEndOf="parent"
                    bind:layout_constraintDimensionRatio="@{model.ratio}"
                    bind:imageURL="@{model.url}"/>
```
I also set  ` precision(Precision.EXACT)
` on Request builder

**Comment by xanscale** [comment]
same problem here
using placeholder break final image size (even with imageview with fixed sizes)

**Comment by Ecgbert** [comment]
I was facing the same problem. 
My solution was to implement a custom imageview to control the aspect ratio and to stop using constraintDimensionRatio.

**Comment by colinrtwhite** [comment]
This is fixed in Coil 2.0.0.
